### DISCUSSION TITLE
Chaching of API Queries

### DISCUSSION THREAD

**Discussion by VincenzoKoestler** [FreshRSS/FreshRSS#6707] [Q&A]
Hey everyone, I realized when using a query like https://domain/api/query.php?user=XY&t=XY=greader&nb=1500 it seems to somehow get cached or at least does not refresh automatically when new articles get added to the feeds? Is there anything I am doing wrong or some setting I am missing? 
There is no data stored in the local browser chache so it seems to be an issue from freshRSS?

Thanks for any ideas and help in resolving this issue :)

**Comment by Alkarex** [comment]
Hello,
How long cache? In which client?

By default, there is a 60s HTTP client cache:
https://github.com/FreshRSS/FreshRSS/blob/51f93e56f7ffc8fbae825124fe740ab623519e0c/p/api/query.php#L165

**Reply by VincenzoKoestler** [reply]
According to the http header the feed was last modified e.g. this morning. 
"last-modified: Wed, 14 Aug 2024 07:52:26 GMT"
So freshRSS does not seem to update the query automatically in the background?
I am not sure the problem also exists on the rss feed or only on the greader feed. 

Edit: Only when I open FreshRSS and interact with an article in the user interface the query gets refreshed
When calling the query the response is 304, but if I interact with an article it changes to 200 for once and goes back to 304

Edit 2: The RSS query seems to return a 200 every time its called

**Reply by Alkarex** [reply]
Ok, I can reproduce the problem, and will try to fix
https://github.com/FreshRSS/FreshRSS/issues/6717
